### DISCUSSION TITLE
Suitable as a data-store for simple applications (like SQLite)?

### DISCUSSION THREAD

**Discussion by adrian5** [duckdb/duckdb#7367] [Q&A] [answered]
I was looking into strongly typed alternatives to SQLite. DuckDB stood out as a modern embedded option. Looking at its presentation, it however appears to be focused on computationally demanding data analytics.

Does that make it overkill as simple data storage format for a small desktop application, or is it nonetheless well suited as a lightweight embedded component (maybe after [taming it](https://duckdb.org/docs/sql/configuration) a bit)?

**Comment by Mytherin** [comment, marked as answer]
I would recommend against using DuckDB's file format as an application file format for now - as the storage format is not yet stabilised and is expected to change each release ([see here](https://duckdb.org/internals/storage)). DuckDB also has a larger code footprint than SQLite as it is much more batteries included - it has a much larger function catalog, far more extensive support for various data types, support for more file formats, support for installing/loading extensions, more extensive SQL syntax, etc. While many of those features can be disabled to strip down the binary size it is not something we support out-of-the-box in a very easy way.

While not a perfect solution perhaps have a look at [SQLite strict tables](https://www.sqlite.org/stricttables.html) which adds type enforcement at the table level to SQLite. That is still not particularly great as far as type safety goes as type safety is not enforced during queries - but it is better than nothing.

You can also use DuckDB's [SQLite extension](https://github.com/duckdblabs/sqlite_scanner) to attach SQLite databases and interact with them in a type-safe manner using DuckDB's execution engine.
